bot.py

```diff
@@ -27,9 +33,14 @@ def load_data():
     return data, model
 
 def fg_make_prob(situation):
-    args = ' '.join("--%s=%r" % (key,val) for (key,val) in situation.iteritems())
+    if sys.version_info[0] >= 3:
+        args = ' '.join("--%s=%r" % (key,val) for (key,val) in situation.items())        
+    else:
+        args = ' '.join("--%s=%r" % (key,val) for (key,val) in situation.iteritems())
     model_fg = muterun_js('model-fg/model-fg.js', args)
-    return model_fg.stdout.split()[-1]
+    stdoutResults = model_fg.stdout
+    stdoutSplit = stdoutResults.split()
+    return stdoutSplit[-1]
 
```

Comment:
Sorry about splitting this line into three -- I separated them for debugging (in order to figure out where some Py3 incorrect behavior was coming from) and forgot to restore to one line.